src/Campaigns.php

```diff
@@ -271,10 +271,18 @@ public function save_taxonomy_meta( $term_id ) {
 		}
 
 		$redirect_page = filter_input( INPUT_POST, 'redirect_page', FILTER_VALIDATE_INT ) ?? 0;
+
 		if ( $redirect_page ) {
 			update_term_meta( $term_id, 'redirect_page', $redirect_page );
+			update_post_meta( $redirect_page, 'redirect_tag_page', $term_id );
```

Comment:
We don't need to add the same information to the post's meta. Instead we should query the tag meta for a tag that has this post set.

Reply:
Would that be really be better? It feels more efficient to set it here rather than query all tags in all pages 🤔

Reply:
Can you elaborate what you mean with "more efficient"?

Reply:
It feels like it would be better performance-wise to just check if that meta attribute is set, rather than loop through all available tags

Reply:
In the end both approaches lead to a SQL query on a table.

Using post meta:
```SQL
select meta_value from wp_postmeta where meta_key='redirect_tag_page' and post_id={{post_id}}
```

Using tag meta:
```SQL
select term_id from wp_termmeta where meta_key='redirect_page' and meta_value={{post_id}}
```

Reply:
It's a query with similar complexity. Even if there's a performance difference, it should be small, as neither query will ever take long.

Keep in mind that our pages are also cached in Cloudflare, and this includes redirects. So even if the difference was significant, it would only affect a tiny portion of page visits.

Reply:
Alright, thanks for clarifying! 😄

Reply:
There's probably a way after all to use a function instead of raw SQL.
https://wordpress.stackexchange.com/questions/231212/get-term-by-custom-term-meta-and-taxonomy

I checked the code and WordPress should fetch the result from cache before using SQL.

https://github.com/WordPress/wordpress-develop/blob/trunk/src%2Fwp-includes%2Fclass-wp-term-query.php#L745-L750